flink-1.17-connector-bigquery/flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/sink/committer/BigQueryCommitter.java

```diff
@@ -30,10 +39,45 @@
  */
 public class BigQueryCommitter implements Committer<BigQueryCommittable>, Closeable {
 
+    private static final Logger LOG = LoggerFactory.getLogger(BigQueryCommitter.class);
+
+    private final BigQueryConnectOptions connectOptions;
+
+    public BigQueryCommitter(BigQueryConnectOptions connectOptions) {
+        this.connectOptions = connectOptions;
+    }
+
     @Override
-    public void commit(Collection<CommitRequest<BigQueryCommittable>> commitRequests)
-            throws IOException, InterruptedException {}
+    public void commit(Collection<CommitRequest<BigQueryCommittable>> commitRequests) {
+        if (commitRequests.isEmpty()) {
+            LOG.info("No committable found. Nothing to commit!");
+            return;
+        }
+        try (BigQueryServices.StorageWriteClient writeClient =
+                BigQueryServicesFactory.instance(connectOptions).storageWrite()) {
+            for (CommitRequest<BigQueryCommittable> commitRequest : commitRequests) {
+                BigQueryCommittable committable = commitRequest.getCommittable();
+                long producerId = committable.getProducerId();
+                String streamName = committable.getStreamName();
+                long streamOffset = committable.getStreamOffset();
+                LOG.info("Committing records appended by producer {}", producerId);
+                FlushRowsResponse response = writeClient.flushRows(streamName, streamOffset);
```

Comment:
What about handling failures and performing retries?

For example, the comparable Dataflow code is here: https://github.com/apache/beam/blob/d93f93abf0e44c050781a635c3c5a834dd292627/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/StorageApiFlushAndFinalizeDoFn.java#L137

Reply:
OK